PWGHF/Tasks/taskMcEfficiency.cxx

```diff
@@ -65,16 +64,24 @@ struct HfTaskMcEfficiency {
 
   void init(o2::framework::InitContext&)
   {
-    auto hCandidates = registry.add<StepTHn>("hCandidates", "Candidate count at different steps", {HistType::kStepTHnF, {axisPt, axisMass, axisPdg, axisCPA, {2, -0.5, 1.5, "collision matched"}}, kHFNSteps});
-    hCandidates->GetAxis(0)->SetTitle("p_{T} (GeV/c)");
-    hCandidates->GetAxis(1)->SetTitle("m_{inv}");
+
+    std::array<bool, 2> doprocessData{doprocessDataD0, doprocessDataLc};
+    std::array<bool, 2> doprocessMc{doprocessMcD0, doprocessMcLc};
+    if (std::accumulate(doprocessData.begin(), doprocessData.end(), 0) > 0 && std::accumulate(doprocessMc.begin(), doprocessMc.end(), 0) > 0) {
+      LOGP(fatal, "Data and MC process functions cannot run simultaneously!");
+    }
+
+    auto hCandidates = registry.add<StepTHn>("hCandidates", "Candidate count at different steps", {HistType::kStepTHnF, {axisPt, axisMass, axisPdg, axisCPA, {2, -0.5, 1.5, "collision matched"}, {RecoDecay::OriginType::NonPrompt + 1, RecoDecay::OriginType::None - 0.5, RecoDecay::OriginType::NonPrompt + 0.5}}, kHFNSteps});
+    hCandidates->GetAxis(0)->SetTitle("#it{p}_{T} (GeV/#it{c})");
+    hCandidates->GetAxis(1)->SetTitle("#it{m}_{inv} (GeV/#it{c}^{2})");
     hCandidates->GetAxis(2)->SetTitle("PDG code");
     hCandidates->GetAxis(3)->SetTitle("CPA");
+    hCandidates->GetAxis(5)->SetTitle("origin");
```

Comment:
Should we call this better? In the end it is prompt or non-prompt, right?

Reply:
We changed in `Charm hadron origin`, which means: `0 = none`, `1 = prompt`, `2 = non-prompt`.